### ISSUE TITLE
check-and-set instead of per-period keys?

### ISSUE THREAD

**Issue by blalor** [EvanKrall/cronsul#4]
I'm wondering if it might be simpler overall to use a check-and-set operation on a key at `/v1/kv/cronsul/${task}/${period}` and have the value be the `${schedule_date}`.  Each node that's running cronsul would read the value and only run the command if the current `schedule_date` is after the key's value _and_ writing the new value with the `cas=` param succeeds.  This would remove the need for `cronsul-cleanup`.

**Comment by EvanKrall** [comment]
Yeah, that sounds like it should work, and I agree that it would be a lot neater.
